File: packages/pdf-generator/android/src/main/java/io/capawesome/capacitorjs/plugins/pdfgenerator/classes/options/GenerateFromUrlOptions.java
package io.capawesome.capacitorjs.plugins.pdfgenerator.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.pdfgenerator.classes.CustomExceptions;

public class GenerateFromUrlOptions extends GenerateOptions {

    @NonNull
    private final String url;

    public GenerateFromUrlOptions(@NonNull PluginCall call) throws Exception {
        super(call);
        this.url = GenerateFromUrlOptions.getUrlFromCall(call);
    }

    @NonNull
    public String getUrl() {
        return url;
    }

    @NonNull
    private static String getUrlFromCall(@NonNull PluginCall call) throws Exception {
        String url = call.getString("url");
        if (url == null) {
            throw CustomExceptions.URL_MISSING;
        }
        return url;
    }
}

File: packages/pdf-generator/ios/Plugin/Classes/Options/GenerateFromUrlOptions.swift
import Foundation
import Capacitor

@objc public class GenerateFromUrlOptions: GenerateOptions {
    let url: URL

    override init(_ call: CAPPluginCall) throws {
        self.url = try GenerateFromUrlOptions.getUrlFromCall(call)
        try super.init(call)
    }

    private static func getUrlFromCall(_ call: CAPPluginCall) throws -> URL {
        guard let urlString = call.getString("url") else {
            throw CustomError.urlMissing
        }
        guard let url = URL(string: urlString) else {
            throw CustomError.urlInvalid
        }
        return url
    }
}
